core/src/com/unciv/models/ruleset/IConstruction.kt

```diff
@@ -45,19 +45,28 @@ interface INonPerpetualConstruction : IConstruction, INamed, IHasUniques {
 
     /** Only checks if it has the unique to be bought with this stat, not whether it is purchasable at all */
     fun canBePurchasedWithStat(city: City?, stat: Stat): Boolean {
+        return canBePurchasedWithStatReasons(city, stat).purchasable
+    }
+
+    /** Only checks if it has the unique to be bought with this stat, not whether it is purchasable at all */
+    fun canBePurchasedWithStatReasons(city: City?, stat: Stat): PurchaseReason {
         val stateForConditionals = StateForConditionals(city?.civ, city)
-        if (stat == Stat.Production || stat == Stat.Happiness) return false
-        if (hasUnique(UniqueType.CannotBePurchased, stateForConditionals)) return false
+        if (stat == Stat.Production || stat == Stat.Happiness) return PurchaseReason.Invalid
+        if (hasUnique(UniqueType.CannotBePurchased, stateForConditionals)) return PurchaseReason.Unpurchasable
         // Can be purchased with [Stat] [cityFilter]
-        if (city != null && getMatchingUniques(UniqueType.CanBePurchasedWithStat, stateForConditionals)
-            .any { it.params[0] == stat.name && city.matchesFilter(it.params[1]) }
-        ) return true
+        if (getMatchingUniques(UniqueType.CanBePurchasedWithStat, StateForConditionals.IgnoreConditionals)
+                .any { city == null ||
+                    (it.conditionalsApply(stateForConditionals) && it.params[0] == stat.name && city.matchesFilter(it.params[1]))
```

Comment:
Params matches stat name should be required even if city is null, this applies to both uniques